src/main.cpp

```diff
@@ -5,49 +5,216 @@
 #include <thread>
 #include <fmt/core.h>
 #include <args.hxx>
-#include <curses.h>
 #include <stdlib.h>
+#include <thread>
+
+// To include the nvml library at runtime
 #ifdef _WIN32
+#define NOMINMAX
 #include <Windows.h>
 #else
 #include <dlfcn.h>
 #endif
 
+// Loading and saving .ini configuration file
 #include "SimpleIni.h"
 #include "setup.hpp"
 
+// Dear ImGui
+#include "imgui.h"
+#include "imgui_impl_glfw.h"
+#include "imgui_impl_opengl3.h"
+#include <GLFW/glfw3.h> // Will drag system OpenGL headers
+
+#pragma region Modifiy InputText so we can use std::string
```

Comment:
This is likely Visual Studio specific. You will get warnings when compiling under Linux.

Nit: Modifiy might be Modify.